File: pset2/readability.c
#include<cs50.h>
#include<stdio.h>
#include<ctype.h>
#include<math.h>

int main(void)
{
    string a=get_string("Text: ");
    float l=0, s=0, w=0, flag=0;
    for(int i=0; a[i]!='\0'; i++)
    {
        if(a[i]==' ')
            flag=0;
        if(isalpha(a[i]))
        {
            if(flag==0)
            {
                w++;
                flag=1;
            }
            l++;
        }
        if(a[i]=='.' || a[i]=='!' || a[i]=='?')
            s++;
    }
    l=(l/w)*100;
    s=(s/w)*100;
    float index=0.0588*l-0.296*s-15.8;
    index=round(index);
    if(index>16)
        printf("Grade 16+\n");
    else if(index<1)
        printf("Before Grade 1\n");
    else
        printf("Grade %i\n", (int)index);
}

File: pset6/readability/readability.py
from cs50 import get_string

a=get_string("Text: ")
l=0
s=0
w=0
flag=0

for element in a:
    if element==' ':
        flag=0
    if element.isalpha():
        if flag==0:
            w+=1
            flag=1
        l+=1
    if element=='.' or element=='!' or element=='?':
        s+=1
        
l=(l/w)*100
s=(s/w)*100

index=0.0588*l-0.296*s-15.8
index=round(index)

if index>16:
    print("Grade 16+")
elif index<1:
    print("Grade 1")
else:
    print(f"Grade {index}")